pkg/plugins/resources/yaml/condition.go

```diff
@@ -18,17 +18,28 @@ func (y *Yaml) Condition(source string) (bool, error) {
 
 // ConditionFromSCM checks if a key exists in a yaml file
 func (y *Yaml) ConditionFromSCM(source string, scm scm.ScmHandler) (bool, error) {
-	if !filepath.IsAbs(y.spec.File) {
-		y.spec.File = filepath.Join(scm.GetDirectory(), y.spec.File)
+	absoluteFiles := make(map[string]string)
+	for filePath := range y.files {
+		absoluteFilePath := filePath
+		if !filepath.IsAbs(filePath) {
+			absoluteFilePath = filepath.Join(scm.GetDirectory(), filePath)
+			logrus.Debugf("Relative path detected: changing to absolute path from SCM: %q", absoluteFilePath)
+		}
+		absoluteFiles[absoluteFilePath] = y.files[filePath]
 	}
+	y.files = absoluteFiles
+
 	return y.condition(source)
 }
 
 func (y *Yaml) condition(source string) (bool, error) {
+	var fileContent string
+	var filePath string
 
-	// Test at runtime if a file exist
-	if !y.contentRetriever.FileExists(y.spec.File) {
-		return false, fmt.Errorf("the yaml file %q does not exist", y.spec.File)
+	if len(y.files) > 1 {
+		validationError := fmt.Errorf("Validation error in conditions of type 'yaml': the attributes `spec.files` can't contain more than one element for conditions")
```

Comment:
Wouldn't it make sense to allow files in condition as well?
To me it only makes sense to disable it for source

Reply:
Let's see what it would look like.
I haven't done this for "type: File", do you think I should too?